sound/soc/sof/intel/hda.c

```diff
@@ -394,17 +423,33 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 			dev_info(bus->dev, "using HDA machine driver %s now\n",
 				 hda_mach->drv_name);
 
-			/* fixup topology file for HDMI only platforms */
-			if (codec_num == 1) {
-				/* use local variable for readability */
-				tplg_filename = pdata->tplg_filename;
-				tplg_filename = fixup_tplg_name(sdev, tplg_filename);
-				if (!tplg_filename) {
-					hda_codec_i915_exit(sdev);
-					return ret;
-				}
-				pdata->tplg_filename = tplg_filename;
+			if (codec_num == 1)
+				idisp_str = "-idisp";
```

Comment:
@plbossart dont you have to allocate memory for idisp_str and dmic_str before assigning?

Reply:
it's just a pointer to a constant string, not sure why memory would need to be allocated.
Actually the variables should be 'const char *'